__tests__/articles.test.js

```diff
@@ -0,0 +1,87 @@
+const app = require("../app.js");
+const request = require("supertest");
+const db = require("../db/connection.js");
+const seed = require("../db/seeds/seed.js");
+const data = require("../db/data/test-data/index.js");
+
+beforeEach(() => {
+  return seed(data);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("/api/articles/:article_id", () => {
+  describe("GET", () => {
+    describe("Status 200", () => {
+      it("should return the correct article given a valid article_id", () => {
+        return request(app)
+          .get("/api/articles/1")
+          .expect(200)
+          .then(({ body }) => {
+            expect(body.article).toHaveProperty("author");
+            expect(body.article).toHaveProperty("title");
+          });
+      });
+
+      it("should return an article with the correct structure", () => {
+        return request(app)
+          .get("/api/articles/1")
+          .expect(200)
+          .then(({ body }) => {
+            expect(body.article).toMatchObject({
+              title: expect.any(String),
+              topic: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+            });
+          });
+      });
+    });
+
+    describe("Status 404", () => {
+      it("should return 404 if article_id does not exist", () => {
+        return request(app)
+          .get("/api/articles/9999")
+          .expect(404)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Article not found");
+          });
+      });
+    });
+
+    describe("Status 400", () => {
+      it("should return 400 for an invalid article_id format", () => {
+        return request(app)
+          .get("/api/articles/not-an-id")
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Invalid input");
+          });
+      });
+    });
+  });
+  describe("Querying with Different Data Types", () => {
+    const invalidInputs = [
+      { type: "floating number", value: "1.23" },
+      { type: "boolean true", value: "true" },
+      { type: "boolean false", value: "false" },
+      { type: "special characters", value: "!@#$%^&*" },
+    ];
+
+    invalidInputs.forEach((input) => {
+      it(`should return 400 for an ${input.type} as article_id`, () => {
+        return request(app)
+          .get(`/api/articles/${input.value}`)
```

Comment:
This is an interesting test, I wouldn't say it's entirely necessary and especially as the values are all `String` values it's testing the same thing 4 times from a data type perspective, a path can only take a string so "true" and "false" won't be read as boolean values by express